xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp

```diff
@@ -142,6 +143,39 @@ bool CGUIDialogSmartPlaylistEditor::OnMessage(CGUIMessage& message)
       HighlightItem(-1);
     }
     break;
+  case GUI_MSG_WINDOW_INIT:
+    {
+      const std::string& startupList = message.GetStringParam(0);
+      if (!startupList.empty())
+      {
+        int party = 0;
+        if (URIUtils::PathEquals(startupList, CProfilesManager::Get().GetUserDataItem("PartyMode.xsp")))
+          party = 1;
+        if (URIUtils::PathEquals(startupList, CProfilesManager::Get().GetUserDataItem("PartyMode-Video.xsp")))
+          party = 2;
+
+        if (party || m_playlist.Load(startupList))
```

Comment:
You still need to load the playlist if it is a partymode playlist. I'd probably go with resetting `m_mode` at the beginning, then setting it to either `partymusic` or `partyvideo` if one of those conditions matches and then after having loaded the playlist, set `m_mode` to `music` or `video` (using the conditions you have in place) if it hasn't been set before.

Does that make sense?